base/deprecated.jl

```diff
@@ -778,6 +778,18 @@ end
 @deprecate cholfact!(A::Base.LinAlg.HermOrSym, uplo::Symbol, ::Type{Val{false}}) cholfact!(A, Val{false})
 @deprecate cholfact!(A::Base.LinAlg.HermOrSym, uplo::Symbol = :U) cholfact!(A)
 
+# #18218
+function arithtype(T)
```

Comment:
Doesn't add `@deprecate` automatically add an export? We probably don't want that here.